reccmp/isledecomp/formats/mz.py

```diff
@@ -0,0 +1,74 @@
+from dataclasses import dataclass
+from pathlib import Path
+import struct
+
+from .image import Image
+
+
+class MZHeaderNotFoundError(ValueError):
+    """MZ magic string not found"""
+
+
+# pylint: disable=too-many-instance-attributes
+@dataclass(frozen=True)
+class ImageDosHeader:
+    # Order is significant!
+    e_magic: int
+    e_cblp: int
+    e_cp: int
+    e_crlc: int
+    e_cparhdr: int
+    e_minalloc: int
+    e_maxalloc: int
+    e_ss: int
+    e_sp: int
+    e_csum: int
+    e_ip: int
+    e_cs: int
+    e_lfarlc: int
+    e_ovno: int
+    e_res: tuple[int, int, int, int]
+    e_oemid: int
+    e_oeminfo: int
+    e_res2: tuple[int, int, int, int, int, int, int, int, int, int]
+    e_lfanew: int
+
+    MAGIC = 0x5A4D  # "MZ"
+
+    @classmethod
+    def from_memory(cls, data: bytes, offset: int) -> tuple["ImageDosHeader", int]:
+        struct_fmt = "<30HI"
+        struct_size = struct.calcsize(struct_fmt)
+        items = struct.unpack_from(struct_fmt, data, offset)
+        result = cls(
+            *items[:14],
+            items[14:18],
+            *items[18:20],
+            tuple[20:30],
+            items[30],
+        )
+        if result.e_magic != cls.MAGIC:
```

Comment:
We should follow the pattern set by the other modules and call `taste()` right at the beginning instead of checking it here